sqlmesh/core/snapshot/evaluator.py

```diff
@@ -944,22 +950,31 @@ def _promote_snapshot(
             )
             adapter.execute(snapshot.model.render_on_virtual_update(**render_kwargs))
 
-        if on_complete is not None:
-            on_complete(snapshot)
+            if (
+                on_complete is not None
+                and snapshots_with_virtual_views
```

Comment:
Can we just do `snapshots_with_virtual_views or set()` upstream and avoid this check here? Also consider using a set instead of a list